### PR TITLE
Kotlin Setup

### PR DESCRIPTION
**Pull request by manijshrestha**
Wecome Kotlin!

### CODE REVIEW

File: app/src/main/kotlin/zeta/android/apps/presenter/NavigationPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package zeta.android.apps.presenter
```
Comment (cdflynn) [inline app/src/main/kotlin/zeta/android/apps/presenter/NavigationPresenter.kt:1]: Kotlin!

File: app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import zeta.android.apps.ui.presentation.NavigationPresentation
+
+
+class NavigationActivity : BaseNavigationActivity(), NavigationPresentation {
+
+    private var mViews: Views? = null
+
+    @Inject lateinit var mPresenter: NavigationPresenter
+
+    internal class Views(root: AppCompatActivity) : BaseViews(root.findViewById(R.id.zeta_drawer_layout)) {
```
Comment (cdflynn) [inline app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt:36]: Whoa, that's neat.  Is the `: BaseViews()` defining the constructor for this class, as opposed to using `constructor //...`?  There's a lot I need to learn about Kotlin.

File: app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    override fun onDestroy() {
+        super.onDestroy()
+        ZetaApplication.watchForMemoryLeaks(applicationContext, this)
+        mPresenter.onDestroy()
+        mViews!!.clear()
+        mViews = null
+    }
+
+    override fun onBackPressed() {
+        if (mViews != null && mViews!!.drawerLayout.isDrawerOpen(GravityCompat.START)) {
```
Comment (cdflynn) [inline app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt]: does it work to do `mViews?.drawerLayout.isDrawerOpen(GravityCompat.START) ?: false`
Pardon my unfamiliarity with the allowed syntax.

File: dependencies.gradle [hunk 4]
```diff
@@ -1,11 +1,11 @@
 ext.versions = [
         compileSdk                        : 25,
-        minSdk                            : 16,
+        minSdk                            : 19,
```
Comment (cdflynn) [inline dependencies.gradle]: 👍

**Review by cdflynn — APPROVED** [review]
@manijshrestha @manjunathc23 
Hope you guys are having fun at IO!  I'm so happy Kotlin is now an official Android language.

Cheers

File: app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt (hunk 2 above)
Comment (manijshrestha) [reply app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt:36]: :) Yea, as time permits I will try to update this project with some kotlin goodies

File: app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt (hunk 3 above)
Comment (manijshrestha) [reply app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt]: yep good catch. I just used the built in "Convert to Kotlin class" from IDE here. it doesn't make sense to do `!!`. I will update this in later commit.

**Comment by manjunathc23** [comment]
@manijshrestha  : Looks like build error :)

**Comment by manjunathc23** [comment]
Should we do another repo just for kotlin? @manijshrestha @cdflynn

File: build.gradle [hunk 5]
```diff
@@ -1,8 +1,5 @@
 import com.android.build.gradle.internal.tasks.AndroidTestTask
 
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
-apply from: 'dependencies.gradle'
```
Comment (manjunathc23) [inline build.gradle]: Is this not required ?

Comment (manijshrestha) [reply build.gradle]: Yep it is no longer required. Sorry haven't had time to update this PR.

File: ci.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+# Please run it from root project directory
+./gradlew clean build -PdisablePreDex -PwithDexcount
```
Comment (manjunathc23) [inline ci.sh]: 👍

**Review by manjunathc23 — COMMENTED** [review]
LGTM

**Comment by manjunathc23** [comment]
@manijshrestha : I am getting lot of rebase conflicts. Can you please rebase this against master whenever you are free?

**Comment by manijshrestha** [comment]
@manjunathc23 , I have already merged master to this branch. Try merging instead of rebasing.

**Comment by manjunathc23** [comment]
This was merged, Thank you @manijshrestha

**Pull request merged**